hal-core/src/mem/page/alloc.rs

```diff
@@ -482,10 +517,13 @@ where
             // Is there a free buddy block at this order?
             if let Some(mut buddy) = unsafe { self.take_buddy(block, curr_order, &mut free_list) } {
                 // Okay, merge the blocks, and try the next order!
+                if buddy < block {
+                    core::mem::swap(&mut block, &mut buddy);
+                }
                 unsafe {
                     block.as_mut().merge(buddy.as_mut());
                 }
-                tracing::trace!("merged with buddy");
+                tracing::trace!("merged with buddy {:?} {:?}", block, buddy);
```

Comment:
nit: prefer structured fields:
```suggestion
                tracing::trace!(?block, ?buddy, "merged with buddy");
```